src/Checkout/AddressForm.js

```diff
@@ -0,0 +1,332 @@
+              }
+            />
+            <span className='flt-label'>City</span>
+          </div>
+        </div>
+
+        <div className='flex-input'>
+          <div className='input'>
+            <textarea
+              className='input-txtarea'
+              cols='20'
+              rows='10'
+              required
+              value={formState.address}
+              onChange={(e) =>
+                formDispatch({ type: "SET_ADDRESS", payload: e.target.value })
+              }
+            ></textarea>
+            <span className='flt-label'>Address</span>
+          </div>
+
+          <div>
+            <select
+              className='input-select'
+              value={formState.state}
+              onChange={(e) =>
+                formDispatch({ type: "SET_STATE", payload: e.target.value })
+              }
+            >
+              {states.map((stte) => {
+                return <option value={stte}>{stte}</option>;
+              })}
+            </select>
+
+            <select
+              className='input-select'
+              value={formState.country}
+              onChange={(e) =>
+                formDispatch({ type: "SET_COUNTRY", payload: e.target.value })
+              }
+            >
+              {countries.map((cntry) => {
+                return <option value={cntry}>{cntry}</option>;
+              })}
+            </select>
+
+            <div className='radio'>
+              <p>Select the Address Type:</p>
+              {addressTypes.map((addType) => {
+                return (
+                  <label className='input-radio-label'>
+                    <input
+                      className='input-radio'
+                      type='radio'
+                      name='address-type'
+                      value={addType}
+                      onChange={(e) =>
+                        formDispatch({
+                          type: "SET_ADDRESSTYPE",
+                          payload: e.target.value,
+                        })
+                      }
+                      checked={formState.addressType === addType}
+                    />
+                    {addType}
+                  </label>
+                );
+              })}
+            </div>
+          </div>
+        </div>
+        <div className='flex-input center-div'>
+          <p>{msg}</p>
+        </div>
+        <div className='flex-input center-div'>
+          <button className='btn  btn-pad primary' type='submit'>
+            {editAddID ? "Save Address" : "Add Address"}
+          </button>
+          <button
+            className='btn  btn-pad secondary'
+            onClick={() => {
+              setAddNewAddress(false);
+              return formDispatch({
+                type: "RESET_FORM",
+                payload: initialFormState,
+              });
+            }}
+            type='reset'
+          >
+            Discard
+          </button>
```

Comment:
The discard button here is not able to do its functionality

Reply:
Fixed! Thank you